dapp/src/hooks/useTransactionFee.ts

```
@@ -1,44 +1,36 @@
 import { useAcreContext } from "#/acre-react/hooks"
-import { logPromiseFailure } from "#/utils"
-import { useEffect, useState } from "react"
-import { ACTION_FLOW_TYPES, ActionFlowType, Fee } from "#/types"
-import { useAppDispatch } from "./store"
+import { queryKeysFactory } from "#/constants"
+import { ACTION_FLOW_TYPES, ActionFlowType, Fees } from "#/types"
+import { useQuery } from "@tanstack/react-query"
 
-export const initialFee = {
-  tbtc: 0n,
-  acre: 0n,
+export const initialFee: Fees = {
+  tbtc: { fee: 0n, isReimbursable: false },
+  acre: { fee: 0n, isReimbursable: false },
   total: 0n,
 }
 
 export default function useTransactionFee(
   amount: bigint | undefined,
   flow: ActionFlowType,
-): Fee {
-  const [depositFee, setDepositFee] = useState<Fee>(initialFee)
+) {
   const { acre } = useAcreContext()
-  const dispatch = useAppDispatch()
 
-  useEffect(() => {
-    if (!amount) {
-      setDepositFee(initialFee)
-    } else {
-      const getEstimatedDepositFee = async () => {
-        if (!acre) return
+  return useQuery({
+    queryKey: [
+      ...queryKeysFactory.userKeys.estimateFee(),
+      flow,
+      amount?.toString(),
+    ],
+    queryFn: () => {
+      if (!acre || !amount) return initialFee
```

I'm not a fan of initial values in Tanstack Query.
It leads to problems with error handling - library doesn't know if error occured, data was missing or anything else.
I would rather throw an error and handle properly.

---

If you decide to add error handling, you could access `isError` flag by calling `useTransactionFee` in `Stake/UnstakeDetails` component and render alternative UI there.
It also justifies the redundancy of `useTransactionDetails` hook I mentioned.

Additionally worth considering implementing loading state by checking `isPending` flag.

If you check both of them, you don't have to worry about missing data (`initialData`). Also TypeScript will catch up with types - no `undefined` by type narrowing they have.